app/src/main/java/org/mozilla/focus/widget/BrowserNavigationOverlay.kt

```diff
@@ -127,6 +128,17 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
         navButtonForward.isFocusable = canGoForward
 
         navUrlInput.setText(navigationProvider?.getCurrentUrl())
+
+        updateFocusOnNavigation()
+    }
+
+    private fun updateFocusOnNavigation() {
```

Comment:
I think requesting focus in the same method where we set focusability is confusing, e.g. if I'm writing new code and I want to update the focused state but not change focus, I'd probably call `updateNavigationButtons` without realizing it might request focus.

For example, one bug I think might exist in your patch is that `onPageStarted` will request focus every time, which may not be desirable, e.g. we have the overlay open, the page moves loads a new URL after some timeout, and then the focus changes inside the overlay.

We should request focus specifically when it's needed, which is only when the drawer is initially shown, if i'm not mistaken.

Reply:
Focus needs to be set every time the drawer is shown, because just making a view VISIBLE does not give it focus.